### PR TITLE
Feature/lxl 3590 see also

### PR DESCRIPTION
**Pull request by olovy**

### CODE REVIEW

**Review by klngwll — APPROVED** [review]
LGTM!

File: whelktool/scripts/cleanups/2022/01/lxl-3590-see-also.groovy [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                }
+            }
+        }
+    }
+}
+
+String find(Map seeAlso) {
+    def query = [
+            'q' : [seeAlso.values().join(" ")]
+    ]
```
Comment (kwahlin) [inline whelktool/scripts/cleanups/2022/01/lxl-3590-see-also.groovy:47]: Clever! I'm stealing this solution for https://jira.kb.se/browse/LXL-3589

File: whelktool/scripts/cleanups/2022/01/lxl-3590-see-also.groovy [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    AtomicReference<String> id = new AtomicReference<>() 
+    selectByIds(queryIds(query).collect()) { candidate -> 
+        def (record, candidateThing) = candidate.graph
+        
+        if (seeAlso.every { key, value -> candidateThing[key] == value}) {
+            id.set(candidateThing.'@id')
+        }
+    }
+    return id.get()
```
Comment (kwahlin) [inline whelktool/scripts/cleanups/2022/01/lxl-3590-see-also.groovy:57]: Can't there be more than one matching agent? Also, maybe more could be matched if variants were compared too?

**Review by kwahlin — COMMENTED** [review]
Nice solution. See comments for possible improvement.

File: whelktool/scripts/cleanups/2022/01/lxl-3590-see-also.groovy (hunk 2 above)
Comment (olovy) [reply whelktool/scripts/cleanups/2022/01/lxl-3590-see-also.groovy:57]: > Can't there be more than one matching agent? 

I think there _shouldn't_ be but there of course will be in the actual data . Good catch. 

> Also, maybe more could be matched if variants were compared too?

Will try compare them with some extra guards.

**Pull request merged**